server/src/main/java/com/coincoinche/lobby/MatchMaker.java

```diff
@@ -0,0 +1,29 @@
+package com.coincoinche.lobby;
+
+import com.coincoinche.websockets.controllers.LobbyController;
+import java.util.LinkedList;
+
+public class MatchMaker {
+  private LobbyController controller;
+  private LinkedList<String> playerQueue;
+
+  public MatchMaker(LobbyController controller) {
+    this.controller = controller;
+    this.playerQueue = new LinkedList<>();
+  }
+
+  public void register(String username) {
+    this.playerQueue.add(username);
+    this.startGameIfPossible();
+  }
+
+  private void startGameIfPossible() {
+    if (playerQueue.size() > 3) {
+      this.playerQueue.removeFirst();
+      this.playerQueue.removeFirst();
+      this.playerQueue.removeFirst();
+      this.playerQueue.removeFirst();
+      this.controller.gameStart();
+    }
```

Comment:
Is this implementation thread-safe? Just wondering, as I guess maybe players will be able to come from multiple threads?

Reply:
I don't think so, and on top of that this first implementation allows only for a single game, because all games events are broadcasted on the same topic and there is no auth!

Reply:
Yeah, true, we'll need to improve this as well in the future.